### PR TITLE
cp: link-deref gnu test fix

### PR DESCRIPTION
**Pull request by matrixhead**
this pr tries to fix gnu test case "link deref" for cp utility 
these are the changes made:

1. when trying to copy a folder without `-R` before throwing "-r not specified" error, cp would make sure that source is not a symlink and noderef option is not given. This way, when both of them are given, it would treat it as a file and copy the symlink itself.

2.  to match gnu cp's behavior, `-R` won't disable dereference if `--link` is given

### CODE REVIEW

File: tests/by-util/test_cp.rs
Comment (BenWiederhake) [inline tests/by-util/test_cp.rs:5295]:
```suggestion
                assert_eq!(src_ino, dest_ino);
```

**Review by BenWiederhake — CHANGES_REQUESTED** [review]
LGTM, except that one typo. (I understand that the typo is also present in the already-existing code; feel free to keep it or fix it as you see fit.)

(Remaining CI failures are known issues #6275 and #6333, so out of scope for this PR.)

Before I merge this, I want to give the other maintainers an opportunity to look at it, because I don't feel 100% confident around symlinks. But this seems  very reasonable.

File: tests/by-util/test_cp.rs
Comment (matrixhead) [reply tests/by-util/test_cp.rs:5295]: okay, I will do it thanks

**Review by BenWiederhake — APPROVED** [review]
LGTM. I'll wait for a few days for the others too look at it, because I don't feel 100% confident around the symlink code.

**Comment by matrixhead** [comment]
is it because I tried to squeeze in deref logic into parsing code

**Comment by sylvestre** [comment]
well done!

**Pull request merged**
